pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -425,3 +388,74 @@ func proxyMapToEnvVars(proxyConfig map[string]string) []corev1.EnvVar {
 	sort.Slice(envVars, func(i, j int) bool { return envVars[i].Name < envVars[j].Name })
 	return envVars
 }
+
+func gracefulTerminationDuration(infrastructureLister configlisterv1.InfrastructureLister) (watchTerminationDuration int) {
```

Comment:
this should go into the config observer too. Any reason we can't do that?

Reply:
do we have access to the config yaml here? We could pick the value from there instead of accessing the infra CR.